go/cmd/dolt/commands/sql.go

```diff
@@ -676,6 +656,12 @@ func runShell(ctx *sql.Context, se *sqlEngine, mrEnv env.MultiRepoEnv) error {
 			}
 		}
 
+		verr := writeRoots(ctx, se, mrEnv, initialRoots)
```

Comment:
You only want to write new roots in the case that err was not nil above.

Should have tests for this.

Reply:
Probably means we need to restructure above logic, since err is only defined in that if block scope.

Reply:
Shouldn't the error be nil for me to write to the roots?

Reply:
Also not sure how to test this, just simulate an error in the shell?